explainaboard/meta_analyses/ranking.py

```diff
@@ -0,0 +1,186 @@
+from typing import Union
+
+import numpy as np
+import pandas as pd
+
+from explainaboard.info import SysOutputInfo
+from explainaboard.meta_analyses.utils import report_to_sysout
+from explainaboard.utils.typing_utils import unwrap
+
+
+class RankingMetaAnalysis:  # (can inherit from an abstract MetaAnalysis class)
```

Comment:
It seems that there is some code redundancy between classes `RankingMetaAnalysis` and `RankFlippingMetaAnalysis`. We can try to abstract these two classes later. Two hints:
* (1) define a base class from which RankingMetaAnalysis and RankFlippingMetaAnalysis can inherit
* (2) define a general class that can cover three situations: 1 system, 2 systems, and multiple systems.